wavesexchange_repos/src/circuit_breaker/mod.rs

```diff
@@ -0,0 +1,249 @@
+pub mod config;
+
+pub use config::Config;
+use wavesexchange_log::debug;
+
+use std::{
+    future::Future,
+    mem::drop,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+use tokio::sync::RwLock;
```

Comment:
What is the justification of using async mutex here? It seems that it is unnecessary.
Also, what is the justification to use RwLock? It seems that every attempt to lock this mutex requires write access, so a regular Mutex would do.

Reply:
We can read-only from rwlock if current and previous queries were successful, so we don't need to obtain write lock and reset state every time.